lib/fluent/plugin/slack_client.rb

```diff
@@ -97,6 +97,35 @@ def with_channels_create(params = {}, opts = {})
           end
         end
       end
+
+      def to_json_with_scrub! (params)
+        retries = 1
+        begin
+          params.to_json
+        rescue Encoding::UndefinedConversionError => e
+          recursive_scrub!(params)
+          if (retries -= 1) >= 0 # one time retry
+            log.warn "out_slack: to_json `#{params}` failed. retry after scrub!. #{e.backtrace[0]} / #{e.message}"
```

Comment:
`e.backtrace[0]` is not necessary because we are sure that the error occurred at L105.